core/chain-configs/src/genesis_config.rs

```diff
@@ -266,6 +286,93 @@ impl GenesisRecords {
     }
 }
 
+impl Default for GenesisRecordsFile {
+    fn default() -> Self {
+        GenesisRecordsFile { path: PathBuf::new(), file_type: GenesisRecordsFileType::FullGenesis }
+    }
+}
+
+/// Visitor for records
+/// Reads records one by one and passes them to sink
+struct RecordsProcessor<F> {
+    sink: F,
+}
+
+impl<'de, F: FnMut(StateRecord)> Visitor<'de> for RecordsProcessor<&'_ mut F> {
+    type Value = ();
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("array of StateRecord")
+    }
+
+    fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
+    where
+        A: SeqAccess<'de>,
+    {
+        while let Some(record) = seq.next_element::<StateRecord>()? {
+            (self.sink)(record)
+        }
+        Ok(())
+    }
+}
+
+impl<'de, F: FnMut(StateRecord)> DeserializeSeed<'de> for RecordsProcessor<&'_ mut F> {
+    type Value = ();
+
+    fn deserialize<D>(self, deserializer: D) -> Result<Self::Value, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        deserializer.deserialize_seq(self)
+    }
+}
+
+/// Visitor for Genesis
+/// Processes records field with records_processor
+/// IGNORES OTHER FIELDS
+struct GenesisRecordsProcessor<F> {
+    records_processor: RecordsProcessor<F>,
+}
+
+impl<'de, F: FnMut(StateRecord)> Visitor<'de> for GenesisRecordsProcessor<&'_ mut F> {
+    type Value = ();
+
+    fn expecting(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        formatter.write_str("map with records field which is array of StateRecord")
+    }
+
+    fn visit_map<A>(self, mut map: A) -> Result<Self::Value, A::Error>
+    where
+        A: MapAccess<'de>,
+    {
+        while let Some(key) = map.next_key::<String>()? {
+            match key.as_str() {
+                "records" => {
+                    map.next_value_seed(self.records_processor)?;
+                    return Ok(());
+                }
+                _ => {
+                    map.next_value::<IgnoredAny>()?;
+                }
+            }
+        }
+        Err(de::Error::custom("missing field: records"))
+    }
+}
+
+pub fn stream_records_from_records_file(reader: impl Read, mut callback: impl FnMut(StateRecord)) {
+    let mut deserializer = serde_json::Deserializer::from_reader(reader);
+    let records_processor = RecordsProcessor { sink: &mut callback };
+    deserializer.deserialize_seq(records_processor).expect("error while processing records");
```

Comment:
I'd return an `serde_json::Result`  out of this function, rather than just unwrapping. It's useful when `.unwraps` are visible at the call-site, especially for such generic and re-usable functions. 

Generally speaking, internal unwraps are fine for such config loading code, but, if propagating errors is not to onerous, one might as well just do it.